internal/user/user.go

```diff
@@ -106,21 +106,42 @@ func Create(ctx context.Context, dbConn *db.DB, cu *CreateUser, now time.Time) (
 }
 
 // Update replaces a user document in the database.
-func Update(ctx context.Context, dbConn *db.DB, userID string, cu *CreateUser, now time.Time) error {
+func Update(ctx context.Context, dbConn *db.DB, id string, upd *UpdateUser, now time.Time) error {
 	ctx, span := trace.StartSpan(ctx, "internal.user.Update")
 	defer span.End()
 
-	if !bson.IsObjectIdHex(userID) {
-		return errors.Wrap(ErrInvalidID, "check objectid")
+	if !bson.IsObjectIdHex(id) {
+		return ErrInvalidID
 	}
 
-	cu.DateModified = &now
-	for _, cua := range cu.Addresses {
-		cua.DateModified = &now
+	fields := make(bson.M)
+
+	if upd.Name != nil {
```

Comment:
Would it make sense to use something like https://github.com/fatih/structs for these mapping lines?

Reply:
No I don't think so. For this class where we want to keep things simple the value that lib provides wouldn't be worth the complexity it would add in terms of learning a new thing. Plus looking at it I don't see that it handles nil fields the way I want.

Reply:
Sounds good!